protocol/strict-derivation.md

```diff
@@ -0,0 +1,355 @@
+in a way to maintain a steady L1/L2 block ratio, e.g. bumping the L1 origin selection every 6 blocks
+in the case of mainnet.
+  - This is similar to Option 1, but rather maintains the current sequencer drift, instead of
+  possibly more eagerly advancing it to get closer to the latest L1 origin.
+  - Edge case: the safe head is already very near the sequencer drift limit, and a newer origin
+  needs to be selected faster. This complicates this solution.
+  - Edge case: L1 missed a slot, and the L1 origin cannot advance as expected. This also needs to be
+  taken care of in Option 1.
+- Option 3: keep last L1 origin as long as possible, so until hitting the sequencing window. This is
+the simplest option, but also means that a sequencer has to fast-advance the L1 origin to catch up
+again. And if the L1 origin already slipped out of the drift window, a sequencer would need to
+continue creating deposit-only blocks until inside the drift window again.
+
+At this point I favor Option 1 because it makes auto-derivation behave similar to a live sequencer.
+Option 2 is quite similar, but gives the sequencer more influence about the "start point" from which
+a steady drift is maintained, which might have unintended consequences. I also see the simplicity of
+Option 3 as an advantage. I would be interested to hear from Proofs/Interop experts which of
+these behaviors play most nicely with Proofs/Interop.
+
+Option 1 (and maybe 2 too) has an interesting side effect: this rule can be used by a sequencer (on
+an almost empty chain) to encode a range of empty batches by just posting a single sort of
+checkpoint batch at the end of the range to then trigger the gap to be auto-derived by these rules.
+For this to fully check out, the sequencer would need a special _gap-sequencing mode_, in which it
+strictly creates blocks according to these rules. It can then safely mark the end of any such gap
+with a single batch.
+
+# Activation rules & actions
+
+I propose to activate the new rules by L1 inclusion block timestamp, which is most convenient and clean
+for derivation protocol upgrades. It's accessible in most parts of the derivation pipeline as the
+stage's _origin_ (not to be confused with the _L1 origin of an L2 block_).
+Note that this is in contrast to how Delta activated span batches, which was by first parsing the
+span batch and then checking the span batch's L1 origin timestamp.
+
+## Frame Queue, Channel Bank, Batch Queue
+
+However, we need to specify how existing buffered frames, channels and batches are treated at
+activation, since the stricter rules of Holocene take effect.
+There are multiple options to enter Holocene with a derivation pipeline state that is consistent
+with Holocene rules.
+
+- Option 1: When the L1 origin reaches the Holocene activation block, discard all frames in the
+  Frame Queue, channels in the Channel Bank, and batches in the Batch Queue.
+  This gives us a nice clean starting point.
```

Comment:
:+1: clean slate during upgrade is nice. We should add some warnings to the batch-submitter about this though, to prevent operational mistakes.